lib/ui/window/platform_configuration.cc

```diff
@@ -396,4 +429,23 @@ std::string PlatformConfigurationNativeApi::DefaultRouteName() {
       ->DefaultRouteName();
 }
 
+int64_t PlatformConfigurationNativeApi::GetRootIsolateToken() {
+  UIDartState* dart_state = UIDartState::Current();
+  FML_DCHECK(dart_state);
+  return dart_state->GetRootIsolateToken();
+}
+
+void PlatformConfigurationNativeApi::RegisterBackgroundIsolate(
+    int64_t isolate_id) {
+  UIDartState* dart_state = UIDartState::Current();
+  FML_DCHECK(dart_state && !dart_state->IsRootIsolate());
+  auto platform_message_handler =
+      (*static_cast<std::shared_ptr<PlatformMessageHandlerStorage>*>(
+          Dart_CurrentIsolateGroupData()));
+  FML_DCHECK(platform_message_handler);
```

Comment:
I'm not quite sure a DCHECK is going to help here.

My concern is that at runtime, during shutdown, something could get messed up and this is nullptr. This will led to a crash during VM shutdown due to a race. Instead I think we should do 
```suggestion
  if (!platform_message_handler) {
    return;
  }
```

Reply:
Did you see my other comment that explains when this could be null?  If the API is used as expected this is guaranteed not to be null, even during the shutdown sequence since the group data outlives the last isolate.  The cases where it could be null are very odd and we'd want to bring attention to them instead of silently suppressing the problem.